rectools/models/sasrec.py

```diff
@@ -561,6 +475,77 @@ def get_dataloader_recommend(self, dataset: Dataset) -> DataLoader:
         return recommend_dataloader
 
 
+# ####  --------------  Lightning Model  --------------  #### #
+
+
+class SasRec(LightningModule):
+    """TODO"""
+
+    def __init__(
+        self,
+        torch_model: TransformerBasedSessionEncoder,
+        lr: float,
+        loss: str = "softmax",  # TODO: move loss func to `SasRec` class
+    ):
+        super().__init__()
+        self.lr = lr
+        self.loss_func = self._init_loss_func(loss)
+        self.torch_model = torch_model
+
+    def forward(
+        self,
+        sessions: torch.Tensor,  # [batch_size, session_maxlen]
+    ) -> torch.Tensor:
+        """TODO"""
+        return self.torch_model(sessions)
+
+    def on_train_start(self) -> None:
+        """TODO"""
+        self.xavier_normal_init()
+
+    def training_step(self, batch: torch.Tensor, batch_idx: int) -> torch.Tensor:
+        """TODO"""
+        x, y, w = batch
+        logits = self.forward(x)  # [batch_size, session_maxlen, n_items + 1]
+        # We are using CrossEntropyLoss with a multi-dimensional case
+
+        # Logits must be passed in form of [batch_size, n_items + 1, session_maxlen],
+        #  where n_items + 1 is number of classes
+
+        # Target label indexes must be passed in a form of [batch_size, session_maxlen]
+        # (`0` index for "PAD" ix excluded from loss)
+
+        # Loss output will have a shape of [batch_size, session_maxlen]
+        # and will have zeros for every `0` target label
+        loss = self.loss_func(logits.transpose(1, 2), y)  # [batch_size, session_maxlen]
+        loss = loss * w
+        n = (loss > 0).to(loss.dtype)
+        loss = torch.sum(loss) / torch.sum(n)
+        return loss
+
+    def configure_optimizers(self) -> torch.optim.Adam:
+        """TODO"""
+        optimizer = torch.optim.Adam(self.torch_model.parameters(), lr=self.lr, betas=(0.9, 0.98))
```

Comment:
Let't move this beta value to this class `__init__` default `adam_betas` parameter